src/ILCompiler.Compiler/src/CppCodeGen/CppWriter.cs

```diff
@@ -974,15 +990,15 @@ public void OutputNodes(IEnumerable<DependencyNode> nodes, MethodDesc entrypoint
                 foreach (var wellKnownTypeNode in _wellKnownTypeNodes)
                 {
                     if (wellKnownTypeNode is EETypeNode)
-                        OutputTypeNode((EETypeNode)wellKnownTypeNode, forwardDefinitions, typeDefinitions, methodTables);
+                        OutputTypeNode(wellKnownTypeNode as EETypeNode, factory, forwardDefinitions, typeDefinitions, methodTables);
```

Comment:
The change to `as EETypeNode` seems unnecessary. We know it's an `EETypeNode` already.